File: problem-solving/baekjoon/1620_나는야_포켓몬_마스터_이다솜.py
# https://www.acmicpc.net/problem/1620
import sys

if __name__ == "__main__":
    n, m = list(map(int, sys.stdin.readline().split()))
    dogam = {}
    for i in range(n):
        name = sys.stdin.readline().rstrip()
        dogam[name] = i+1
        dogam[str(i+1)] = name
    for i in range(m):
        question = sys.stdin.readline().rstrip()
        print(dogam[question])

File: problem-solving/baekjoon/1620_나는야_포켓몬_마스터_이다솜.go
// https://www.acmicpc.net/problem/1620
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n, m int
	fmt.Fscanln(reader, &n, &m)

	var dogam = map[string]string{}
	for i := 0; i < n; i++ {
		var name string
		fmt.Fscanln(reader, &name)
		dogam[name] = fmt.Sprintf("%d", i+1)
		dogam[fmt.Sprintf("%d", i+1)] = name
	}
	for i := 0; i < m; i++ {
		var input string
		fmt.Fscanln(reader, &input)
		fmt.Fprintln(writer, dogam[input])
	}
}
